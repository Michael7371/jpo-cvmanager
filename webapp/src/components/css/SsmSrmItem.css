#ssmitemdiv {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  text-align: center;
  padding: 10px;
  border-bottom: 1px #0e2052 solid;
  background: #fff1d3;
}

#srmitemdiv {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  text-align: center;
  padding: 10px;
  border-bottom: 1px #0e2052 solid;
  background: #bdceff;
}

.btnActive {
  height: fit-content;
  width: fit-content;
  font-family: Arial, Helvetica, sans-serif;
  padding: 5px;
  color: white;
  background: #d16d15;
  border-color: white;
  border-radius: 5px;
  cursor: pointer;
  font-weight: bold;
}

.btnDisabled {
  height: fit-content;
  width: fit-content;
  font-family: Arial, Helvetica, sans-serif;
  padding: 5px;
  color: white;
  background: #888888;
  border-color: white;
  border-radius: 5px;
  font-weight: bold;
}

.ssmsrmitemtext {
  font-family: Arial, Helvetica, sans-serif;
  color: #0e2052;
  width: 90px;
  padding: 5px;
}
