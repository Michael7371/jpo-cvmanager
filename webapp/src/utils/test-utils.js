/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

export { replaceChaoticIds }
