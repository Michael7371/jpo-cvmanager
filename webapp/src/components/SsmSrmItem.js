import React, { useEffect } from 'react'

import './css/SsmSrmItem.css'

const SsmSrmItem = (props) => {
  const { setSelectedSrm } = props
  useEffect(() => {
    return () => {
      setSelectedSrm({})
    }
  }, [setSelectedSrm])

  return (
    <div id={props.elem['type'] === 'srmTx' ? 'srmitemdiv' : 'ssmitemdiv'}>
      <p className="ssmsrmitemtext">{props.elem['time']}</p>
      <p className="ssmsrmitemtext">{props.elem['requestId']}</p>
      <p className="ssmsrmitemtext">{props.elem['role']}</p>
      <p className="ssmsrmitemtext">{props.elem['status']}</p>
      {props.elem['type'] === 'srmTx' ? (
        <button className="btnActive" onClick={() => props.setSelectedSrm(props.elem)}>
          View
        </button>
      ) : (
        <button className="btnDisabled" disabled={true}>
          View
        </button>
      )}
    </div>
  )
}

export default SsmSrmItem
