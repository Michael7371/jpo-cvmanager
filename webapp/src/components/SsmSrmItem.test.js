import React from 'react'
import { render } from '@testing-library/react'
import SsmSrmItem from './SsmSrmItem'
import { replaceChaoticIds } from '../utils/test-utils'

it('should take a snapshot', () => {
  const { container } = render(<SsmSrmItem elem={{}} setSelectedSrm={() => {}} />)

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
