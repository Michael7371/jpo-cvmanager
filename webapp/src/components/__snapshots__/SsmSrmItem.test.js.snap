// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    id="ssmitemdiv"
  >
    <p
      class="ssmsrmitemtext"
    />
    <p
      class="ssmsrmitemtext"
    />
    <p
      class="ssmsrmitemtext"
    />
    <p
      class="ssmsrmitemtext"
    />
    <button
      class="btnDisabled"
      disabled=""
    >
      View
    </button>
  </div>
</div>
`;
